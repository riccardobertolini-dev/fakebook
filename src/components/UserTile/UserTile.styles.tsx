import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';

const commonMargin = css`
	padding-bottom: 15px;
`;

export const SocialWallTile = styled.div`
	border: 1px solid #ccc;
	background-color: #fff;
	border-radius: 10px;
	padding: 20px 10px;
	margin-bottom: 20px;
`;

export const UserInfo = styled.div`
	display: flex;
	align-content: center;
	align-items: center;
	gap: 10px;
	font-weight: 300;
	font-family: 'Lato', sans-serif;
	font-size: 14px;
	${commonMargin};

	img {
		border-radius: 50%;
		width: 40px;
		height: 40px;
		object-fit: contain;
	}
`;

export const ImageContainer = styled.div`
	position: relative;
	overflow: hidden;
	max-height: 400px;
`;

export const ImagePosted = styled.img`
	width: 100%;
	transition: 0.3s;
	object-fit: contain;

	&:hover {
		transform: scale(1.1);
	}
`;

export const Quote = styled.div`
	font-weight: 400;
	font-size: 18px;
	font-family: 'Lato', sans-serif;
	${commonMargin};
`;

export const SocialInteractions = styled.div`
	display: flex;
	position: relative;
	gap: 2px;
`;

const InteractionButtonsDesign = css`
	width: 50%;
	height: 40px;
	font-size: 16px;
	font-family: 'Lato', sans-serif;
	font-weight: 700;
	background-color: #f3f3f3;
	border: 1px solid #eee;
	border-radius: 5px;
`;

export const LikeButton = styled.button`
	${InteractionButtonsDesign};
	cursor: pointer;
`;

export const CommentButton = styled(Link)`
	${InteractionButtonsDesign};
	display: flex;
	align-items: center;
	justify-content: center;
	color: black;
	text-decoration: none;
`;
