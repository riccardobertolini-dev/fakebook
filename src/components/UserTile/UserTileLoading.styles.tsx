import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';

const commonMargin = css`
	padding-bottom: 15px;
`;

export const SocialWallTile = styled.div`
	border: 1px solid #ccc;
	background-color: #fff;
	border-radius: 10px;
	padding: 20px 10px;
	margin-bottom: 20px;
`;

export const UserInfo = styled.div`
  display: flex;
  align-content: center;
  align-items: center;
  gap: 10px;
  font-weight: 300;
  font-family: 'Lato', sans-serif;
  font-size: 14px;
  ${commonMargin};
  
  span {
    display: block;
	background-color: #ccc;
	width: 40px;
	height: 14px;
    &:nth-child(2) {

	  width: 60px;
	}
  }
}`;

export const EmptyProfileImage = styled.div`
	border-radius: 50%;
	width: 40px;
	height: 40px;
	background-color: dimgrey;
	object-fit: contain;
`;

export const ImageContainer = styled.div`
	position: relative;
	overflow: hidden;
	background-color: dimgrey;
	height: 400px;
`;

export const QuoteLoading = styled.div`
	font-weight: 400;
	font-size: 18px;
	width: 100%;
	height: 20px;
	margin-bottom: 20px;
	background-color: #ccc;
	font-family: 'Lato', sans-serif;
	${commonMargin};
`;

export const SocialInteractions = styled.div`
	display: flex;
	position: relative;
	gap: 2px;
`;

const InteractionButtonsDesign = css`
	width: 50%;
	height: 40px;
	font-size: 16px;
	font-family: 'Lato', sans-serif;
	font-weight: 700;
	background-color: #f3f3f3;
	border: 1px solid #eee;
	border-radius: 5px;
`;

export const LikeButton = styled.button`
	${InteractionButtonsDesign};
	cursor: pointer;
`;

export const CommentButton = styled(Link)`
	${InteractionButtonsDesign};
	display: flex;
	align-items: center;
	justify-content: center;
	color: black;
	text-decoration: none;
`;
