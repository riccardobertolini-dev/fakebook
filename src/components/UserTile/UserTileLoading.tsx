import { ImageContainer, UserInfo, LikeButton, QuoteLoading, SocialInteractions, SocialWallTile, EmptyProfileImage } from './UserTileLoading.styles';

type LoadingProps = {
	times: number;
};

export const UserTileLoading = ({ times }: LoadingProps) => {
	return (
		<>
			{[...Array(times)].map(() => (
				<SocialWallTile>
					<UserInfo>
						<EmptyProfileImage />
						<span></span> <span></span>
						<span></span>
					</UserInfo>
					<QuoteLoading></QuoteLoading>
					<ImageContainer></ImageContainer>
					<SocialInteractions>
						<LikeButton></LikeButton>
						<LikeButton></LikeButton>
					</SocialInteractions>
				</SocialWallTile>
			))}
		</>
	);
};
