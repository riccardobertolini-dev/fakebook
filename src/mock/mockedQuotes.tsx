export const MockedQuote1 = {
	text: 'Genius is one percent inspiration and ninety-nine percent perspiration.',
	author: 'Thomas Edison',
};
export const MockedQuote2 = {
	text: 'You can observe a lot just by watching.',
	author: 'Yogi Berra',
};
