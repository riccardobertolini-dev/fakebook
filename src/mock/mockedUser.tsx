export const MockedUser1 = {
	gender: 'female',
	name: {
		title: 'Ms',
		first: 'Isabelle',
		last: 'Davies',
	},
	location: {
		street: {
			number: 8255,
			name: 'Esk Street',
		},
		city: 'Taupo',
		state: 'West Coast',
		country: 'New Zealand',
		postcode: 25207,
		coordinates: {
			latitude: '53.3834',
			longitude: '16.3516',
		},
		timezone: {
			offset: '-6:00',
			description: 'Central Time (US & Canada), Mexico City',
		},
	},
	email: 'isabelle.davies@example.com',
	login: {
		uuid: '71d3d4bf-2c47-450f-b3bf-e15a7fef23da',
		username: 'yellowtiger203',
		password: 'pyramid',
		salt: 'WrEua7np',
		md5: '8d185b252bdc6739cc36a7d4c67bcb1f',
		sha1: '4843e0607dd75b952504e278013508847f002237',
		sha256: 'dde6119b6dad7740579c5ace7f89789b7c24973dd203285bac2028018a046926',
	},
	dob: {
		date: '1965-10-08T18:06:10.878Z',
		age: 57,
	},
	registered: {
		date: '2015-12-03T15:58:10.458Z',
		age: 6,
	},
	phone: '(999)-128-6616',
	cell: '(132)-372-7000',
	id: {
		name: '',
		value: null,
	},
	picture: {
		large: 'https://randomuser.me/api/portraits/women/44.jpg',
		medium: 'https://randomuser.me/api/portraits/med/women/44.jpg',
		thumbnail: 'https://randomuser.me/api/portraits/thumb/women/44.jpg',
	},
	nat: 'NZ',
};
