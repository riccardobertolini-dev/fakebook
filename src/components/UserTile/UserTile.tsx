import { useEffect, useState } from 'react';
import { User } from '../../models/User';
import { QuoteProps } from '../../models/Quotes';
import { SocialWallTile, UserInfo, ImagePosted, Quote, ImageContainer, SocialInteractions, LikeButton, CommentButton } from './UserTile.styles';
import { getQuotes } from '../../api/getQuote';
import { AiOutlineLike, AiFillLike } from 'react-icons/ai';

type UserProps = {
	user: User;
	index: number;
	selfContent: boolean;
	hardCodedquote?: string | null;
};

export function UserTile({ user, index, selfContent, hardCodedquote = null }: UserProps) {
	const [quote, updateQuote] = useState<QuoteProps>();
	const [liked, isLiked] = useState<boolean>(false);
	const [likesNumber, updateLikes] = useState<number>(0);

	useEffect(() => {
		const fetchQuote = async () => {
			await getQuotes().then(result => {
				updateQuote(result);
			});
		};

		fetchQuote();

		updateLikes(Math.floor(Math.random() * 100));
	}, []);

	const userLikedPost = () => {
		isLiked(!liked);
		updateLikes(liked ? likesNumber - 1 : likesNumber + 1);
	};

	return (
		<SocialWallTile>
			<UserInfo>
				<img src={user.picture.medium} alt="" />
				{user.name.first} {user.name.last}
			</UserInfo>
			{selfContent ? <Quote>{hardCodedquote}</Quote> : <Quote>{(quote && quote[index]?.text) || ''}</Quote>}

			{!selfContent && (
				<ImageContainer>
					<ImagePosted src={`https://picsum.photos/400/450?random=${user.dob.age}`} />
				</ImageContainer>
			)}
			{!selfContent && (
				<SocialInteractions>
					<LikeButton onClick={() => userLikedPost()}>
						{liked ? <AiFillLike /> : <AiOutlineLike />}
						{likesNumber} Like(s)
					</LikeButton>
					<CommentButton to={`/${user.login.uuid}`}>Open Profile</CommentButton>
				</SocialInteractions>
			)}
		</SocialWallTile>
	);
}
