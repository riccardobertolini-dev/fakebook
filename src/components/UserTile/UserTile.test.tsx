import * as React from 'react';
import renderer from 'react-test-renderer';
import { UserTile } from './UserTile';
import { MockedUser1 } from '../../mock/mockedUser';

jest.mock('axios');

jest.mock('react-router-dom', () => ({
	Routes: jest.fn(),
	Route: jest.fn(),
	Link: jest.fn(),
}));

describe('UserTile component', () => {
	it('should renders', () => {
		const wrapper = renderer.create(<UserTile user={MockedUser1} index={0} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
