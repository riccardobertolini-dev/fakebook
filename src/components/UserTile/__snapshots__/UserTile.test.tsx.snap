// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTile component should renders 1`] = `
<div
  className="sc-bcXHqe kdCfmY"
>
  <div
    className="sc-gswNZR ievbtS"
  >
    <img
      alt=""
      src="https://randomuser.me/api/portraits/med/women/44.jpg"
    />
    Isabelle
     
    Davies
  </div>
  <div
    className="sc-eDvSVe hzsoTC"
  />
</div>
`;
